rllib/policy/sample_batch.py

```diff
@@ -93,11 +93,17 @@ def __init__(self, *args, **kwargs):
 
         lengths = []
         copy_ = {k: v for k, v in self.items()}
+        batch_size = 0
         for k, v in copy_.items():
             assert isinstance(k, str), self
             len_ = len(v) if isinstance(
                 v,
                 (list, np.ndarray)) or (torch and torch.is_tensor(v)) else None
+            # Observation data is always in the buffer dict.
+            if k == SampleBatch.OBS:
+                batch_size = len(v) if isinstance(
```

Comment:
what about batch_size = len_
the code is identical. My mistake was assuming that OBS is always the first one in the list lengths, that's the mistake. Sorry about that

Reply:
Ah my silly code, perhaps bit tired today. updated now.